### PR TITLE
Create ipv4_conversion.py

### LINKED ISSUE #9718: Speedup our eight slowest pytests (one at a time please)
### Feature description

At the end of our GitHub Actions `build` jobs, there is a list of the slowest ones.

Are there ways to speed up these tests without reducing our functionality or our code coverage?

Please only fix one algorithm per pull request.
* [ ] 16.64s call     neural_network/simple_neural_network.py::neural_network.simple_neural_network.forward_propagation
* [x] 6.15s call     backtracking/word_search.py::backtracking.word_search.word_exists
* [x] 4.61s call     digital_image_processing/test_digital_image_processing.py::test_local_binary_pattern
* [x] 2.59s call     backtracking/power_sum.py::backtracking.power_sum.backtrack
* [x] 2.19s call     machine_learning/xgboost_regressor.py::machine_learning.xgboost_regressor.main
* [x] 1.73s call     maths/prime_numbers.py::maths.prime_numbers.slow_primes
* [x] 1.43s call     backtracking/power_sum.py::backtracking.power_sum.solve
* [x] 1.00s call     other/dijkstra_bankers_algorithm.py::other.dijkstra_bankers_algorithm.BankersAlgorithm.main
================= 1506 passed, 25 warnings in 71.23s (0:01:11) =================

Also, those **25 pytest warnings** are worth fixing!!!

### PR DESCRIPTION
**Pull request by Shocker-lov-t**
### Describe your change:

Closes #9718
Closes #11013

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: conversions/ipconversion.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    Returns:
+        int: The decimal representation of the IP address.
+
+    >>> ip_to_decimal("192.168.0.1")
+    3232235521
+    >>> ip_to_decimal("10.0.0.255")
+    167772415
+    """
+
+    ip_parts = ip_address.split(".")
```
Comment (cclauss) [inline conversions/ipconversion.py:20]: See: https://github.com/TheAlgorithms/Python/blob/master/maths/is_ip_v4_address_valid.py
```suggestion
    octets = ip_address.split(".")
```

File: conversions/ipconversion.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    Returns:
+        str: The IPv4 representation of the decimal IP address.
+
+    >>> decimal_to_ip(3232235521)
+    '192.168.0.1'
+    >>> decimal_to_ip(167772415)
+    '10.0.0.255'
+    """
+
+    if not (0 <= decimal_ip <= 4294967295):
```
Comment (cclauss) [inline conversions/ipconversion.py:47]: This would not catch 10.0.0.256 which is not a valid IP.

Comment (Shocker-lov-t) [reply conversions/ipconversion.py:47]: Thanks for the Valuable catch.

**Review by Shocker-lov-t — COMMENTED** [review]
Thanks for the Valuable Catches.

File: conversions/ipconversion.py (hunk 1 above)
Comment (Shocker-lov-t) [reply conversions/ipconversion.py:20]: Using more Appropriate variable names indeed make the code easy to understand Thanks!.

File: conversions/ipconversion.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    ip_parts = ip_address.split(".")
+    if len(ip_parts) != 4:
+        raise ValueError("Invalid IPv4 address format")
+
+    decimal_ip = 0
+    for part in ip_parts:
+        decimal_ip = (decimal_ip << 8) + int(part)
+
+    return decimal_ip
```
Comment (cclauss) [inline conversions/ipconversion.py:28]:
```
def alt_ip_to_decimal(ip_address: str) -> int:
    return int("0x" + "".join(f"{int(i):02x}" for i in ip_address.split(".")), 16)
```

**Review by cclauss — APPROVED** [review]
Nice one!

**Pull request merged**